instrumentation/reactor-3.1/library/src/main/java/io/opentelemetry/instrumentation/reactor/ReactorTracing.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.reactor;
+
+import io.opentelemetry.context.Context;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class ReactorTracing {
+
+  private static final Object TRACE_CONTEXT_KEY =
+      new Object() {
+        @Override
+        public String toString() {
+          return "otel-trace-context";
+        }
+      };
+
+  static volatile Mono<String> dummyMono;
+  static volatile Flux<String> dummyFlux;
+
+  static synchronized void enable() {
```

Comment:
Also it's probably good to merge the public API of `TracingOperator` into this class if I'm not mistaking anything. We just want one public entry point for reactor context propagation. I guess then a user would do `ReactorContextPropagation.enable()` or `ReactorContextPropagation.disable()`, those methods would call into TracingOperator, instead of other way around.

Reply:
Oh I had missed trask's comment, it seems to overlap. 

> Yeah, I can do it. I thought that as we'll add more tracing stuff (start/end span convenience) having configuration (register/reset hooks) on the same class may get noisy, so I thought about separating tracing.

I agree with this - but my understanding is the operator is about context propagation and not creating spans, so it would still be appropriate to merge in